lib/cartopy/tests/mpl/test_gridliner.py

```diff
@@ -242,8 +246,9 @@ def test_grid_labels_tight():
 
     # Ensure gridliners were drawn
     for ax in fig.axes:
-        for gl in ax._gridliners:
-            assert hasattr(gl, '_drawn') and gl._drawn
+        for artist in ax.artists:
+            if isinstance(artist, Gridliner):
+                assert hasattr(artist, '_drawn') and artist._drawn
```

Comment:
I realize this was not your doing, but this seems fragile if something changes types and we don't hit this code path. Could we change it to a count assertion `num_gridliners_drawn == expected_number` to make sure we are hitting the code path we expect?